Use the new HLG pack/unpack API in Dask

madsbk opened the pull request:
This PR makes Distributed use the new high level graph pack/unpack API in Dask: https://github.com/dask/dask/pull/7179

- [ ] Tests added / passed
- [ ] Passes `black distributed` / `flake8 distributed`

ian-r-rose commented on distributed/scheduler.py:
```diff
@@ -3619,8 +3618,13 @@ def update_graph_hlg(
     ):
-
-        dsk, dependencies, annotations = highlevelgraph_unpack(hlg, annotations)
+        (
```
Holy autoformatting, batman

madsbk replied:
Fixed :)

ian-r-rose reviewed (commented):
:+1:

jrbourbeau reviewed (commented):
Thanks for your work here @madsbk! I pushed an update to run CI against https://github.com/dask/dask/pull/7179 to ensure that the distributed test suite passes too.

We'll also want to update the minimum supported version of `dask` to make sure `HighLevelGraph`s have the new `__dask_distributed_(un)pack__` API. Historically what we've done in this case is update `distributed`s `requirements.txt` to point to the `master` branch of `dask` and add a `TODO` comment to update to the latest `dask` release prior to releasing `distributed`

madsbk commented:
Thanks @jrbourbeau, both PRs should work now. I have added some tests of the issue in Dask and renamed `anno => annotations` here.

Some of the CI runs fails in `test_register_backend_entrypoint`, it doesn't seem related to this PR?

jrbourbeau commented:
> Some of the CI runs fails in test_register_backend_entrypoint, maybe related to the CI pointing to Dask PR dask/dask#7179 ?

Yeah you're correct, that failure is be related to pointing CI to your branch. Since our versioning system, `versioneer`, uses commit tags to determine the package version, and your branch most recent tag is `1.2.2`, we're not meeting some minimum requirement in the `dask` version that test relies on.

```
pkg_resources.VersionConflict: (dask 1.2.2+1025.g9d91e6a0 (/usr/share/miniconda3/envs/dask-distributed/lib/python3.6/site-packages), Requirement.parse('dask>=2021.02.0'))
```

This is okay as it's only an artifact of pointing to a fork of `dask`

jrbourbeau reviewed (approved):
Thanks @madsbk! This LGTM. See my comment at https://github.com/dask/dask/pull/7179#pullrequestreview-599185244 about waiting a day to merge this PR
